lib/index.js

```diff
@@ -324,10 +324,29 @@ function walk (opts, nodes) {
       // run the loop, different types of loops for arrays and objects
       if (Array.isArray(target)) {
         for (let index = 0; index < target.length; index++) {
+          opts.locals.loop = {
+            index: index,
+            remaining: target.length - index - 1,
+            first: target.indexOf(target[index]) === 0,
+            last: index + 1 == target.length,
+            length: target.length
+          }
```

Comment:
I think it can be put in a separate method `getLocalsLoop`

Reply:
You mean `opts.locals.loop = getLocalsLoop()`?

Reply:
yes, like this may be 

```js
const getLocalsLoop = (index, target) => ({
  index: index,
  remaining: target.length - index - 1,
  first: target.indexOf(target[index]) === 0,
  last: index + 1 == target.length,
  length: target.length
});
```

and then use it like this 

`opts.locals.loop = getLocalsLoop(index, target)`

Reply:
Cool, exactly how I imagined it, too 👍